charts/loki-distributed/templates/gateway/service-gateway.yaml

```diff
@@ -17,8 +17,16 @@ spec:
   {{- with .Values.gateway.service.clusterIP }}
   clusterIP: {{ . }}
   {{- end }}
-  {{- if and (eq "LoadBalancer" .Values.gateway.service.type) .Values.gateway.service.loadBalancerIP }}
-  loadBalancerIP: {{ .Values.gateway.service.loadBalancerIP }}
+  {{- if eq "LoadBalancer" .Values.gateway.service.type }}
+  {{- if .Values.gateway.service.loadbalancerip }}
+  loadbalancerip: {{ .Values.gateway.service.loadbalancerip }}
+  {{- end }}
+  {{- if .Values.gateway.service.loadBalancerSourceRanges  }}
```

Comment:
nit: WDYT of using a with here?

Reply:
Hi @trevorwhitney, thx a lot.

When I develop my own Service objects always use range (like Prometheus chart: https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus/service.yaml#L29-L33), but I checked grafana chart Helm (https://github.com/grafana/helm-charts/blob/main/charts/grafana/templates/service.yaml#L27-L30) and use `toYaml .Values.service.loadBalancerSourceRanges | indent 4`. IMHO, result will be the same.

Reply:
yeah, I don't think we need a range, but was suggesting a `with` instead of the `if`.